stores/AzureRecorderActions.ts

```diff
@@ -16,34 +17,45 @@ export const startRecording = async (router: NextRouter) => {
   clearTimeout(get().recorderTimeout);
   const { apiKeyAzure, apiKeyAzureRegion } = get();
 
-  let lastTextUpdate = "";
+  let textUpdates: string[] = [];
 
-  const updateText = (text: string, persist: boolean) => {
-    // Remove previous text update
-    const textInputValue = get().textInputValue;
-    const cutText = textInputValue.substring(
-      0,
-      textInputValue.length - lastTextUpdate.length
-    );
+  const persistText = (text: string) => {
+    const effectiveInputValue = `${get().textInputValue} `;
     set((state) => ({
-      textInputValue: `${cutText}${text}${persist ? " " : ""}`,
+      textInputValue: effectiveInputValue,
     }));
-    // Reset if persisting, otherwise update
-    if (persist) {
-      lastTextUpdate = "";
-      if (get().autoSendStreamingSTT) {
-        if (!get().activeChatId) {
-          addChat(router);
-        }
-        submitMessage({
-          id: uuidv4(),
-          content: text,
-          role: "user",
-        });
-        set((state) => ({ textInputValue: "" }));
+    if (get().autoSendStreamingSTT) {
+      if (!get().activeChatId) {
+        addChat(router);
       }
-    } else {
-      lastTextUpdate = text;
+      submitMessage({
+        id: uuidv4(),
+        content: effectiveInputValue,
+        role: "user",
+      });
+      set((state) => ({ textInputValue: "" }));
+      textUpdates = [];
+    }
+  };
+  const { submit_debounce_ms: submitDebounce  } = get().settingsForm;
+  const debouncedPersistText = debounce(persistText, submitDebounce);
+
+  const updateText = (text: string, recognitionComplete: boolean) => {
+    text = text.trim();
+    set((state) => ({
+      textInputValue: `${textUpdates.join(" ")} ${text}`,
+    }));
+    if (recognitionComplete) {
+      textUpdates.push(text);
+    }
+  };
+
+  const updateAndEventuallyPersistText = (text: string, recognitionComplete: boolean) => {
+    updateText(text, recognitionComplete);
+    if (submitDebounce) {
+      debouncedPersistText(text);
+    } else if (recognitionComplete) {
+      persistText(text);
     }
   };
```

Comment:
Splitting up UI updates and message submissions into different operations so we can configure how each works distinct from the other.